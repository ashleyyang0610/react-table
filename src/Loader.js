import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleNotch } from '@fortawesome/free-solid-svg-icons';

const Loader = ({ ...props }) => {
    return (
        <FontAwesomeIcon
            icon={faCircleNotch}
            size="3x"
            spin
            style={{
                color: '#ced4da',
                ...props.style
            }}
        />
    );
};

export default Loader;
