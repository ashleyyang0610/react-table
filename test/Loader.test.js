import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Loader from '../src/Loader';

describe('Loader', () => {
    let loader;

    describe('with default settings', () => {
        beforeEach(() => {
            loader = shallow(<Loader />);
        });

        test('should render correctly', () => {
            expect(toJSON(loader)).toMatchSnapshot();
        });
    });
});
