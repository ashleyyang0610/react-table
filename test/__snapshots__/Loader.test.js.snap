// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader with default settings should render correctly 1`] = `<styled.div />`;
